src/Lean/Meta/Tactic/Simp/Main.lean

```diff
@@ -402,41 +380,400 @@ def simpForall (e : Expr) : SimpM Result := withParent e do
   else
     return { expr := (← dsimp e) }
 
+/--
+Information about a single `have` in a `have` telescope.
+Created by `getHaveTelescopeInfo`.
+-/
+structure HaveInfo where
+  /-- Previous `have`s that the type of this `have` depends on. -/
+  typeBackDeps : Std.HashSet Nat
```

Comment:
@kmill Are the `Nat` here the de-Bruijn indices of the variables? If yes, let's document it.